code/modules/psionics/faculties/coercion.dm

```diff
@@ -60,8 +60,9 @@
 		to_chat(user, SPAN_WARNING("\The [target] is in no state for a mind-read."))
 		return TRUE
 
-	var/obj/item/organ/internal/augment/psi/psiaug = locate() in target
-	if (!psiaug?.is_broken())
+	var/mob/living/carbon/C = target
+	var/obj/item/organ/internal/augment/psi/psiaug = C.internal_organs_by_name[BP_AUG_PSI]
+	if(psiaug && !psiaug.is_broken())
 		if(target.isSynthetic())
 			to_chat(user, SPAN_ALIEN("There is no mind here for you to dip your mentality into."))
```

Comment:
This codeblock and the block on lines 267+ are essentially the same thing. It could be moved into a separate proc, especially if you expect that this proc might be used by other psionic powers in the future. 

The check for if a mob has a psi-receiver could live in `code\modules\psionics\mob\mob.dm` 
The messages if the target is a certain kind of alien can go somewhere else. Maybe it can be defined under the psi receiver definition.